scripts/core/editor3/store/index.ts

```diff
@@ -125,11 +129,17 @@ export const getDecorators = (options: IOptions) => {
         );
     }
 
-    if (limitConfig?.ui === 'highlight' && typeof limitConfig?.chars === 'number') {
+    const isHardHighlight = limitConfig?.ui === 'highlight' && typeof limitConfig?.chars === 'number';
+    const isSoftHighlight = softLimitConfig?.ui === 'highlight' && typeof softLimitConfig?.chars === 'number';
+
+    if (isHardHighlight) {
```

Comment:
It appears that soft limit will only work if hard limit is configured a certain way. They should be independent.